src/main/kotlin/com/chattriggers/ctjs/api/render/Gui.kt

```diff
@@ -452,6 +453,42 @@ class Gui @JvmOverloads constructor(
         }
     }
 
+    /**
+     * Sets the button's text
+     *
+     * @param buttonId id of the button
+     * @param text the new text
+     */
+    fun setButtonText(buttonId: Int, text: TextComponent) = apply {
+        buttons[buttonId]?.message = text
+    }
+
+    /**
+     * Sets the button's text
+     *
+     * @param buttonId id of the button
+     * @param text the new text
+     */
+    fun setButtonText(buttonId: Int, text: String) = setButtonText(buttonId, TextComponent(text))
+
+    /**
+     * Sets the gui's tooltip, this will be visible on top of the cursor
+     * when the gui is open.
+     *
+     * @param text the contents of the tooltip
+     */
+    fun setTooltip(text: TextComponent) = apply {
+        setTooltip(Tooltip.wrapLines(Client.getMinecraft(), text))
+    }
```

Comment:
This one already exists on `Screen`

Reply:
It isn't a public method though, that's why I included it

Reply:
Ahh, missed that